subgraphs/enzyme-core/mappings/v4/ExternalPositionManager.ts

```diff
@@ -1809,6 +1824,165 @@ export function handleCallOnExternalPositionExecutedForFund(event: CallOnExterna
     return;
   }
 
+  if (type.label == 'PENDLE_V2') {
+    if (actionId == PendleV2ActionId.BuyPrincipalToken) {
+      let decoded = ethereum.decode(
+        '(address,address,uint256,tuple(uint256,uint256,uint256,uint256,uint256),uint256)',
+        event.params.actionArgs,
+      );
+
+      if (decoded == null) {
+        return;
+      }
+
+      let tuple = decoded.toTuple();
+
+      let market = tuple[0].toAddress();
+      let depositToken = ensureAsset(tuple[1].toAddress());
+      let depositAmount = toBigDecimal(tuple[2].toBigInt(), depositToken.decimals);
+
+      let pendleMarket = usePendleV2AllowedMarket(Address.fromString(vault.id), market);
+
+      let position = usePendleV2Position(event.address.toHex());
+      position.principalTokenHoldings = arrayUnique(position.principalTokenHoldings.concat([pendleMarket.id]));
+      position.save();
+
+      let change = createPendleV2PositionChange(event.params.externalPosition, 'BuyPrincipalToken', vault, event);
+      change.assetAmounts = [
+        createAssetAmount(depositToken, depositAmount, denominationAsset, 'pendle-buy-pt', event).id,
+      ];
```

Comment:
could we include also `assets` array? In other integration, we also add `assets` even if they are included in `assetAmounts`. So `assetAmounts` is like a subgroup of `assets`. Same for the other actions